### PR TITLE
[Doc] Update jedis integration doc and move to new integrations section

### LINKED ISSUE #4322: [Doc:] Helidon MP CDI documentation
Helidon MP CDI documentation (according to the new documentation structure).

Path: `/HelidonMP/CDI`

May include `Extensions` subchaper.

- [x] Overview (may be reused from current doc)
- [x] Maven Dependency (may be reused from current doc)
- [x] Usage (may be reused from current doc and guide)
- [x] API (may be reused from current doc)
- [x] Configuration (may be reused from current doc)
- [x] Reference (generated)
- [x] Examples (may be reused from guide)
- [x] Additional Information

### PR DESCRIPTION
**Pull request by barchetta**
Also deletes MP Extensions section which is now empty.

Resolves #4322

### CODE REVIEW

File: docs/mp/integrations/jedis.adoc [hunk 1]
```diff
@@ -62,19 +74,26 @@ The following examples show how to create and inject a Jedis pool named `orders`
...
-The extension implements this injection point by creating a
- link:{jedis-base-url}/jedis/JedisPool.html[JedisPool] object in the
- link:{jakarta-cdi-javadoc-url}/javax/enterprise/context/ApplicationScoped.html[application scope].
+Helidon implements this injection point by creating a
+link:{jedis-base-url}/jedis/JedisPool.html[JedisPool] object in the
+link:{jakarta-cdi-javadoc-url}/jakarta/enterprise/context/ApplicationScoped.html[application scope].
 
 You can configure the object using xref:../jaxrs/server-configuration.adoc[MicroProfile config].
- For example, the Jedis pool created above can be configured as follows:
+For example, the Jedis pool created above can be configured as follows:
```
Comment (arjav-desai) [inline docs/mp/integrations/jedis.adoc:82]: These two lines are duplicate i.e. its in configuration section below too.

File: docs/mp/integrations/jedis.adoc [hunk 2]
```diff
@@ -62,19 +74,26 @@ The following examples show how to create and inject a Jedis pool named `orders`
...
+
+== Configuration
+
+You can configure the object using xref:../jaxrs/server-configuration.adoc[MicroProfile config].
```
Comment (arjav-desai) [inline docs/mp/integrations/jedis.adoc:86]: Nit: make xref fully qualified using {rootdir} or {docdir}

File: docs/mp/integrations/jedis.adoc [hunk 3]
```diff
@@ -157,3 +176,8 @@ Injection without a `@Named` annotation is also possible:
 ----
 In this case, the properties for JedisPoolConfig and JedisPool that can be set will start wih `redis.clients.jedis.JedisPoolConfig.default` and `redis.clients.jedis.JedisPool.default` respectively.
```
Comment (arjav-desai) [inline docs/mp/integrations/jedis.adoc:177]: typo: with

File: docs/mp/integrations/jedis.adoc (hunk 2 above)
Comment (ljnelson) [inline docs/mp/integrations/jedis.adoc:86]: Is that the correct `.adoc`?

File: docs/mp/integrations/jedis.adoc [hunk 4]
```diff
@@ -16,17 +16,28 @@
...
+
+- <<Overview, Overview>>
+- <<maven-coordinates, Maven Coordinates>>
+- <<Usage, Usage>>
+- <<Configuration, Configuration>>
+- <<References, References>>
+
+== Overview
+
+link:https://github.com/xetorthio/jedis[Jedis] is a Java client for link:https://github.com/redis/redis[Redis].
```
Comment (ljnelson) [inline docs/mp/integrations/jedis.adoc:37]: The Jedis Github link here does not match the one at the end of this document.

File: docs/mp/integrations/jedis.adoc [hunk 5]
```diff
@@ -157,3 +176,8 @@ Injection without a `@Named` annotation is also possible:
 ----
 In this case, the properties for JedisPoolConfig and JedisPool that can be set will start wih `redis.clients.jedis.JedisPoolConfig.default` and `redis.clients.jedis.JedisPool.default` respectively.
 
+== References
+
+* link:{helidon-github-tree-url}/examples/integrations/cdi/jedis[Helidon Jedis Usage Examples]
+* link:{jedis-base-url}/jedis/JedisPool.html[JedisPool] javadoc
+* link:https://github.com/redis/redis[Redis]
```
Comment (ljnelson) [inline docs/mp/integrations/jedis.adoc:183]: This doesn't match the Jedis Github link at the top of this document.

File: docs/mp/integrations/jedis.adoc (hunk 1 above)
Comment (barchetta) [reply docs/mp/integrations/jedis.adoc]: fixed

File: docs/mp/integrations/jedis.adoc (hunk 3 above)
Comment (barchetta) [reply docs/mp/integrations/jedis.adoc]: fixed

File: docs/mp/integrations/jedis.adoc (hunk 2 above)
Comment (barchetta) [reply docs/mp/integrations/jedis.adoc]: Fixed

File: docs/mp/integrations/jedis.adoc (hunk 5 above)
Comment (barchetta) [reply docs/mp/integrations/jedis.adoc:182]: Fixed

File: docs/mp/integrations/jedis.adoc (hunk 4 above)
Comment (barchetta) [reply docs/mp/integrations/jedis.adoc:37]: Fixed

File: docs/mp/integrations/jedis.adoc (hunk 2 above)
Comment (barchetta) [reply docs/mp/integrations/jedis.adoc]: Fixed

**Review by arjav-desai — APPROVED** [review]
LGTM

**Review by ljnelson — APPROVED** [review]

**Review by ljamen — COMMENTED** [review]
Changes look good to me.

**Pull request merged**